extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElement, useRef, useState} from '../../../src/preact';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import {useIntersect} from './use-intersect';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  const {0: timestamp, 1: setTimestamp} = useState(
+    getFuzzyTimestampValue(
+      props['datetime'],
+      props['locale'],
+      props['init'],
+      props['cutoff']
+    )
+  );
+  const ref = useRef(null);
+  const entries = useIntersect(ref);
+  const last =
+    entries.length > 0 ? entries[entries.length - 1] : {isIntersecting: false};
+  const {isIntersecting} = last;
+  if (isIntersecting) {
+    setTimestamp(
```

Comment:
Interesting, I didn't realize it would re-render when it comes in and out of viewport.